SerialPrograms/Source/Integrations/DiscordSocial/DiscordSocial.cpp

```diff
@@ -0,0 +1,170 @@
+                    state = "Running"; 
+                    m_activity.SetAssets(assets); break;
+                }
+                case ProgramState::STOPPED:{
+                    assets.SetSmallImage("stopped");
+                    assets.SetSmallText("Stopped");
+                    state = "Stopped";
+                    m_activity.SetAssets(assets); break;
+                }
+                default: {
+                    assets.SetSmallImage("idle");
+                    assets.SetSmallText("Idle");
+                    state = "Idle";
+                    m_activity.SetAssets(assets); break;
+                }
+            }
+
+            if (item.second.program_name != details){
+                details = item.second.program_name;
+            }
+
+            m_timestamps.SetStart(std::chrono::duration_cast<std::chrono::seconds>(item.second.start_time.time_since_epoch()).count());
+        }
+
+        m_activity.SetTimestamps(m_timestamps);
+        m_activity.SetDetails(details);
+        m_activity.SetState(state);
+        m_client->UpdateRichPresence(m_activity, [&](ClientResult result){
+            if (!result.Successful()){
+                log("Rich Presence update failed.", "update_rp()", LoggingSeverity::Warning);
+            }
+        });
+    }catch (const std::exception& e){
+        log("Exception: " + std::string(e.what()), "update_rp()", LoggingSeverity::Error);
+        m_running.store(false, std::memory_order_release);
+    }
+}
+
+Logger& DiscordSocial::logger(){
+    static TaggedLogger logger(global_logger_raw(), "DiscordSocialSDK");
+    return logger;
+}
+
+void DiscordSocial::log(const std::string& message, const std::string& identity, const LoggingSeverity& severity){
+    if (severity < m_log_level){
+        return;
+    }
+
+    std::string log = identity + ": " + message;
+    Color color;
+    switch (severity){
+        case LoggingSeverity::Verbose: color = COLOR_GRAY; break;
+        case LoggingSeverity::Info: color = COLOR_CYAN; break;
+        case LoggingSeverity::Warning: color = COLOR_ORANGE; break;
+        case LoggingSeverity::Error: color = COLOR_MAGENTA; break;
+        default: color = COLOR_PURPLE; break;
+    };
+
+    logger().log(log, color);
+}
+
+Activity DiscordSocial::create_activity(){
+    Activity activity{};
+    activity.SetType(ActivityTypes::Playing);
+    activity.SetDetails("ComputerControl: Preparing to automate...");
+    activity.SetState("Idle");
+
+    ActivityButton button1{};
+    button1.SetLabel("Download from GitHub");
+    button1.SetUrl("https://github.com/PokemonAutomation/ComputerControl/releases/latest");
+    activity.AddButton(button1);
+
+    ActivityButton button2{};
+    button2.SetLabel("Join our Discord");
+    button2.SetUrl("https://discord.gg/pokemonautomation");
+    activity.AddButton(button2);
+
+    discordpp::ActivityAssets assets{};
+    assets.SetLargeImage("rotom");
+    assets.SetLargeText("Pok\u00E9mon Automation");

```

Comment:
More strings you can pull here: https://github.com/PokemonAutomation/Arduino-Source/blob/main/SerialPrograms/Source/CommonFramework/Globals.cpp#L46